logoutHandler.js

```diff
@@ -0,0 +1,39 @@
+import AWS from "aws-sdk";
+import util from "util";
+const { promisify } = util;
+
+AWS.config.update({
+  accessKeyId: process.env.AWS_ACC_KEY_ID,
+  secretAccessKey: process.env.AWS_SECRET
+});
+const COGNITO_CLIENT = new AWS.CognitoIdentityServiceProvider({
+  apiVersion: "2016-04-19",
+  region: "us-east-1"
+});
+const utilPromiseAdminLogout = promisify(
+  COGNITO_CLIENT.adminUserGlobalSignOut
```

Comment:
from my reading, this will invalidate all tokens and log the user out of all devices (so if they were using the same auth on their phone they would have to log in again). given we only store our tokens for a session anyway, not longer term like you would with a mobile app, this is likely fine for us.... but something to consider.

Reply:
That's what I thought as well that all tokens are invalidated however upon some testing, found out that it wasn't the case.

Our middleware runs jwt.verify so as long as the token hasn't expired or if it's a proper token. It will pass regardless if user had been signed out. At the moment I am trying to see if we can run another check like get Current user session off cognito

Reply:
Upon further investigation found this: 

> Hello,

> You can control the duration of token by specifying TokenDuration in GetOpenIdTokenForDeveloperIdentity. By default it's valid for 15 minutes but you can extend it up to 24 hours.

> But once the token has been issued, it can not be terminated. Although your application can choose to discard the existing token locally at any point, depending on your use case. You can then force the re authentication of a user and refresh the credentials with new token.


https://forums.aws.amazon.com/thread.jspa?threadID=232652

Reply:
Resolved it with:

https://dzone.com/articles/aws-cognito-user-pool-access-token-invalidation-1